doc/src/Howto_bulk2slab.rst

```diff
@@ -0,0 +1,138 @@
+===========================
+Convert bulk system to slab
+===========================
+
+A regularly encountered simulation problem is how to convert a bulk
+system that has been run for a while to equilibrate into a slab system
+with some vacuum space.  The challenge here is that one cannot just
+change the box dimensions with the :doc:`change_box command
+<change_box>` or edit the box boundaries in a data file because some
+atoms will have non-zero image flags from diffusing around.  Changing
+the box dimensions results in an undesired displacement of those atoms,
+since the image flags indicate how many times the box length in x-, y-,
+or z-direction needs to be added or subtracted to get the "unwrapped"
+coordinates.  By changing the box dimension this distance is changed and
+thus those atoms move unphysically relative to their neighbors with zero
+image flags.  Setting image flags forcibly to zero creates problems because
+that could break apart molecules by have one atom of a bond on the top
+of the system and the other at the bottom.
+
+.. _bulk2slab:
+.. figure:: JPG/rhodo-both.jpg
+   :figwidth: 80%
+   :figclass: align-center
+
+   Snapshots of the bulk Rhodopsin system (right) and the slab geometry (left)
+
+Below is a suggested workflow using the :doc:`Rhodopsin benchmark input
+<Speed_bench>` for demonstration.  The figure shows the state before on
+the left (with unwrapped atoms that have diffused out of the box) and
+after on the right (with the vacuum added above and below).  The process
+is done by modifying the ``in.rhodo`` input file.  The first lines up to
+and including the :doc:`read_data command <read_data>` remain unchanged.
+Then we insert the following lines to add vacuum to the z direction
+above and below the system:
+
+.. code-block:: LAMMPS
+
+   variable       delta index 10.0
+   reset_atoms    image all
+   write_dump     all custom rhodo-unwrap.lammpstrj id xu yu zu
+   change_box     all z final $(zlo-2.0*v_delta) $(zhi+2.0*v_delta) &
+                      boundary p p f
+   read_dump      rhodo-unwrap.lammpstrj 0 x y z box no replace yes
+   kspace_modify  slab 3.0
+
+Specifically, the :doc:`variable delta <variable>` (set to 10.0)
+represents a distance that determines the amount of vacuum added: we add
+twice its value in each direction to the z-dimension; thus in total
+:math:`40 \AA` get added.  The :doc:`reset_atoms image all
+<reset_atoms>` command shall reset any image flags to become either 0 or
```

Comment:
Comment: The command `reset_atoms image all` is very important for the trickery here. To create the 2 interfaces, I think the value of `delta` would need to be sufficiently large compared to the molecule dimensions (e.g. end-to-end distance, or 2 times the radius of gyration). Of course, this is system specific: short chains like in the example, or really long chains wrapping around the original box multiple times.

Reply:
I guess I need to add a disclaimer, that this will not work for all systems, especially polymers where chains could be connected across periodic boundaries. Those cannot be converted to slabs. There may be other pathological cases.

Reply:
done